jose-jwt/jwe/JWE.cs

```diff
@@ -0,0 +1,330 @@
+﻿#if NETSTANDARD
+namespace Jose.Jwe
+{
+    using Newtonsoft.Json;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+
+    public class Recipient
+    {
+        /// <summary
+        /// A recipient for a Jwe
+        /// </summary
+        /// <param name="alg">algorithm to be used to encrypt the CEK (Content Encryption Key).</param>
+        /// <param name="key">key for encrypting CEK (Content Encryption Key). Cannot be null.</param>
+        public Recipient(JweAlgorithm alg, object key, IDictionary<string, object> perRecipientHeaders = null)
+        {
+            this.Alg = alg;
+            this.Key = key ?? throw new ArgumentNullException(nameof(key));
+            this.PerRecipientHeaders = perRecipientHeaders;
+        }
+
+        public JweAlgorithm Alg { get; set; }
+
+        public object Key { get; set; }
+
+        internal IDictionary<string, object> PerRecipientHeaders { get; }
+    }
+
+    public enum SerializationMode
+    {
+        smCompact,

```

Comment:
Probably just `Compact` and `Json`? `sm` prefix looks so 1990ish :)

Reply:
Sure - done.